packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -665,15 +665,22 @@ export class Project extends ProjectBase {
       encryptionKey: props.encryptionKey && props.encryptionKey.keyArn,
       badgeEnabled: props.badge,
       cache,
-      name: props.projectName,
+      name: new cdk.Token(() => this.tryPhysicalName).toString(),
       timeoutInMinutes: props.timeout,
       secondarySources: new cdk.Token(() => this.renderSecondarySources()),
       secondaryArtifacts: new cdk.Token(() => this.renderSecondaryArtifacts()),
       triggers: this.source.buildTriggers(),
     });
 
-    this.projectArn = resource.projectArn;
-    this.projectName = resource.ref;
+    this.projectArn = new cdk.CrossAccountRegionPhysicalArnToken(resource.projectArn, new cdk.Token(() => this.node.stack.formatArn({
+      service: 'codebuild',
+      resource: 'project',
+      resourceName: this.physicalName,
+    })), this).toString();
+    this.projectName = new cdk.CrossAccountRegionPhysicalNameToken(resource.projectName, this).toString();
```

Comment:
I don't understand why this has to be a different class from `CrossAccountRegionPhysicalArnToken`. Is the difference that the second one knows that `obj.physicalName` exists, but the first one doesn't? Because that doesn't seem strong enough of a reason to make two classes to me, especially given that the second one seems like it could be written as:

```ts
new cdk.CrossEnvironmentIdentifier(resource.projectName, () => this.physicalName).toString();
```